### PR TITLE
e2e: improve description of framework callbacks

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

When Ginkgo shows a BeforeEach/AfterEach/DeferCleanup, then it can only show the source code where the callback was registered because there is no description parameter. This can be improved by passing a custom CodeLocation.

Because a description like "set up framework" might not be enough, the source code is still shown, too.

#### Which issue(s) this PR fixes:
Related-to: https://github.com/onsi/ginkgo/issues/1023

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/internal/unittests/cleanup/cleanup_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ import (
 //
 // This must be line #50.
 
-var _ = ginkgo.Describe("framework", func() {
+var _ = ginkgo.Describe("e2e", func() {
```
Comment (pohly) [inline test/e2e/framework/internal/unittests/cleanup/cleanup_test.go:52]: This gets changed to make it more obvious that this is unrelated to the "framework" in the annotated code location.

**Comment by oomichi** [comment]
The ginkgo version has been updated already with 311144fb3e61dabdbcd48d164dc80716b5ba5a73 for using the feature.
This seems straightforward for me, thanks.

/test pull-kubernetes-integration
/test pull-kubernetes-e2e-gce-100-performance
/lgtm

**Comment by aojea** [comment]
:heart:
